### ISSUE TITLE
WithError method on Logger

### ISSUE THREAD

**Issue by caiorcferreira** [americanas-go/log#28]
With the current interface we can log the error in a structured fashion as:
```go
log.WithField("err", err.Error()).Error("something bad happened")
```

This is a pattern that I face often, hence I think it would be nice to have a syntax sugar for the above line, for example:
```go
// the line below is equal to log.WithField("err", err.Error()).Error("something bad happened")
log.WithError(err).Error("something bad happened")
```

The `Logger` could also accept an option to customize the field name to be used in this method.

**Comment by jpfaria** [comment]
Just don't forget. You have to implement in the 3 implementations (zap, zerolog and logrus).

**Comment by caiorcferreira** [comment]
Issue solved by PR #29
